Fixes #722 - Handle two sizes images

karlcow opened the pull request:
This modifies the python upload code and the JavaScript libraries handling the markup.
Note that for the JS, the previous code is using two different methods for inserting the code at the right place even if they do exactly the same thing. This should be abstracted. 

r? @miketaylr

miketaylr commented on webcompat/static/js/lib/bugform.js:
```diff
@@ -428,8 +428,10 @@ function BugForm() {
+   this.addImageURL = function(url) {
+    var file_ext = url.split('.').pop();
+    var thumb_url = url.replace('.' + file_ext, '-thumb.' + file_ext);
+    var imageURL = ['[![Screenshot Description](', thumb_url, ')](', url, ')'].join('');
```
Here we're doing string concatenation manually, but in issues.js we're using Lodash's `_.template` method. I don't have a preference (which makes since I authored the original two different versions... :/), but can we make them consistent?

karlcow replied:
yes that was the purpose of my comment in the pull request. :) I tried to minimize my modification of the JS. And I remember you telling me a couple of months ago that to avoid to repeat the same code, we should probably make a function which is called in the two places. 

I have 0 preferences on the way to do it given my understanding of JS (performance impact, elegance, etc.) The one without Lodash's template seems more direct?

miketaylr replied:
> yes that was the purpose of my comment in the pull request

Gotcha, yeah there's duplication (in spirit) now, but let's make it real duplication (until we fix that).

karlcow replied:
done :)

miketaylr commented on webcompat/static/js/lib/bugform.js:
```diff
@@ -428,8 +428,10 @@ function BugForm() {
-    var imageURL = ['![Screenshot Description](', url, ')'].join('');
+   this.addImageURL = function(url) {
+    var file_ext = url.split('.').pop();
+    var thumb_url = url.replace('.' + file_ext, '-thumb.' + file_ext);
```
Aren't we getting back the `thumb_url` path back from the response sent back by the `@uploads` `/` route?

Why not just make `addImageURL` take a response argument, rather than the url, that way you can grab `response.url` and `response.thumb_url` without the string manipulation.

miketaylr replied:
nit: Can we also update the comment to mention the thumbnail?

karlcow replied:
done.

miketaylr commented on webcompat/static/js/lib/issues.js:
```diff
@@ -199,8 +199,11 @@ issues.ImageUploadView = Backbone.View.extend({
-    var compiledImageURL = imageURL({url: response.url});
+    var img_url = response.url;
+    var file_ext = img_url.split('.').pop();
+    var thumb_url = img_url.replace('.' + file_ext, '-thumb.' + file_ext);
```
Same as before, we should be able to just get `response.thumb_url` here, right?

karlcow replied:
done :)

karlcow commented:
ok addressed all the comments. And also pushed up a bit the size of the image and changed the algorithm for compression, so screenshots of texts are usable on desktop.

NEAREST at the top, BILINEAR at the bottom.
![capture d ecran 2017-01-06 a 09 39 03](https://cloud.githubusercontent.com/assets/505230/21703357/c4e0cb74-d3f4-11e6-922d-d5542e7dc9e3.png)

karlcow commented:
Thanks @miketaylr for the cool review. :)

miketaylr commented:
Looks good! r+

(Can you file an issue for adding some thumbnail tests?)

karlcow commented:
ahah. forgotten about that. Done with #1267

❤️

miketaylr commented:
Let's revert this for now, sorry @karlcow!

Another thing to fix when you figure out why the bug form uploads are broken: if you click "Remove image upload", it should remove the link in description.

See https://github.com/webcompat/webcompat.com/blob/master/webcompat/static/js/lib/bugform.js#L383-L385
